src/main/java/net/unicon/lti/controller/lti/OIDCController.java

```diff
@@ -137,6 +137,7 @@ public String loginInitiations(HttpServletRequest req, HttpServletResponse res,
                 if (StringUtils.isBlank(parameters.get("ltiStorageTarget"))){
                     return "redirect:" + parameters.get("oicdEndpointComplete");
                 } else {
+                     model.addAllAttributes(parameters);
```

Comment:
I added this because the HTML template needed access to more data (nonce, login hint, client id, lti message hint, oicd endpoint), but let me know if this should be something else!

Reply:
I was curious about this too, will this duplicate the params added below?

Reply:
model basically acts like a Hash in Ruby (it actually extends a couple different implementations of HashMap) - it will simply overwrite the previously set variables (with the same values).